core/src/OpenMP/Kokkos_OpenMP_Parallel.hpp

```diff
@@ -167,55 +216,83 @@ class ParallelFor<FunctorType, Kokkos::MDRangePolicy<Traits...>,
   inline static void exec_range(const MDRangePolicy& mdr_policy,
                                 const FunctorType& functor, const Member ibeg,
                                 const Member iend) {
-#ifdef KOKKOS_ENABLE_AGGRESSIVE_VECTORIZATION
-#ifdef KOKKOS_ENABLE_PRAGMA_IVDEP
-#pragma ivdep
-#endif
-#endif
+    KOKKOS_PRAGMA_IVDEP_IF_ENABLED
     for (Member iwork = ibeg; iwork < iend; ++iwork) {
       iterate_type(mdr_policy, functor)(iwork);
     }
   }
 
+  inline static void exec_work(const MDRangePolicy& mdr_policy,
+                               const FunctorType& functor, const Member iwork) {
+    iterate_type(mdr_policy, functor)(iwork);
+  }
+
+  template <class Policy>
+  typename std::enable_if<std::is_same<typename Policy::schedule_type::type,
+                                       Kokkos::Dynamic>::value>::type
+  execute_parallel() const {
+#pragma omp parallel for schedule(dynamic KOKKOS_OPENMP_OPTIONAL_CHUNK_SIZE) \
+    num_threads(OpenMP::impl_thread_pool_size())
+    KOKKOS_PRAGMA_IVDEP_IF_ENABLED
+    for (auto iwork = m_policy.begin(); iwork < m_policy.end(); ++iwork) {
+      exec_work(m_mdr_policy, m_functor, iwork);
+    }
+  }
+
+  template <class Policy>
+  typename std::enable_if<!std::is_same<typename Policy::schedule_type::type,
+                                        Kokkos::Dynamic>::value>::type
+  execute_parallel() const {
+#pragma omp parallel for schedule(static KOKKOS_OPENMP_OPTIONAL_CHUNK_SIZE) \
+    num_threads(OpenMP::impl_thread_pool_size())
+    KOKKOS_PRAGMA_IVDEP_IF_ENABLED
+    for (auto iwork = m_policy.begin(); iwork < m_policy.end(); ++iwork) {
+      exec_work(m_mdr_policy, m_functor, iwork);
+    }
+  }
+
  public:
   inline void execute() const {
-    enum {
-      is_dynamic = std::is_same<typename Policy::schedule_type::type,
-                                Kokkos::Dynamic>::value
-    };
-
     if (OpenMP::in_parallel()) {
       ParallelFor::exec_range(m_mdr_policy, m_functor, m_policy.begin(),
                               m_policy.end());
-    } else {
-      OpenMPExec::verify_is_master("Kokkos::OpenMP parallel_for");
+      return;
+    }
+
+    OpenMPExec::verify_is_master("Kokkos::OpenMP parallel_for");
+#ifndef KOKKOS_INTERNAL_DISABLE_NATIVE_OPENMP
```

Comment:
Where is this defined?

Reply:
This is not defined anywhere - it's intended as an advanced option for developers (so you'd have to go with `add_definitions(-DKOKKOS_INTERNAL_DISABLE_NATIVE_OPENMP)` or similar). This was requested not to be exposed as public option, but a temporary fallback.

Reply:
Ah, OK.